extensions/in-memory/did-document-store-inmem/src/main/java/org/eclipse/dataspaceconnector/samples/identity/did/InMemoryDidDocumentStore.java

```diff
@@ -88,5 +89,18 @@ public Instant getCreateTime() {
         public int compareTo(Entity<T> other) {
             return createTime.compareTo(other.getCreateTime());
         }
+
+        @Override
+        public boolean equals(Object o) {
```

Comment:
nit: is this triggered because the class implements Comparable? It does not seem correct that this class implements Comparable just to sort by date. It would seem correct to have the class not implement Comparable. (Other classes can implement a Comparator if they want to achieve such sorting).

Reply:
Comparable should be used to reflect natural ordering, in this case it's chronological order. Moreover, it's private nested class so there are no other classes that are going to use it except InMemoryDidDocumentStore.